### PR TITLE
docs(tutorial): hugo with github actions

### PR DESCRIPTION
**Pull request by bene2k1**
### Description
Tutorial Hugo on Object Storage with GitHub Actions

### CODE REVIEW

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+categories: 
+  - storage
+  - object-storage
+dates:
+  validation: 2022-12-07
+  posted: 2022-12-07
+---
+
+Hugo is a static HTML and CSS website generator written in Go. The application is designed to take a directory with content and templates and renders them into a full HTML website.
+It uses Markdown files with front matter for metadata making it lightweight and easily to use.
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:18]:
```suggestion
It uses Markdown files with front matter for metadata, making it lightweight and easy to use. 
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  - object-storage
+dates:
+  validation: 2022-12-07
+  posted: 2022-12-07
+---
+
+Hugo is a static HTML and CSS website generator written in Go. The application is designed to take a directory with content and templates and renders them into a full HTML website.
+It uses Markdown files with front matter for metadata making it lightweight and easily to use. 
+
+In this tutorial you learn how to install Hugo on your local computer so you can build your website locally, before pushing your content into a GitHub repository where a GitHub runner will take the content, generate the HTML version of it and uploads on a Object Storage bucket automatically.
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:20]:
```suggestion
In this tutorial you learn how to install Hugo on your local computer so you can build your website locally, before pushing your content into a GitHub repository where a GitHub runner will take the content, generate the HTML version of it and upload to a Object Storage bucket automatically.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have an Object Storage bucket with [bucket website feature](/storage/object/how-to/use-bucket-website) enabled
+  - You have a [GitHub](https://github.com/) account and [created](https://docs.github.com/en/get-started/quickstart/create-a-repo) an empty repository for your project
+  - You have `git` installed on your local computer
+</Message>
+
+## Installing Hugo
+
+Hugo is available for most operating systems. In this tutorial we describe the installation on a Unbuntu Linux system. For other operating systems, refer to the [official documentation](https://gohugo.io/installation/).
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:33]:
```suggestion
Hugo is available for most operating systems. In this tutorial we describe the installation on a Ubuntu Linux system. For other operating systems, refer to the [official documentation](https://gohugo.io/installation/).
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    [[deployment.targets]]
+    # Configure the deployment target to the Object Storage bucket. Make sure your bucket is named exactly as your base URL (mysuperwebsite.example.com)
+    name = "mysuperwebsite.example.com"
+    URL = "s3://mysuperwebsite.example.com?endpoint=https://s3.fr-par.scw.cloud&region=fr-par"
+    ```
+    <Message type="tip">
+        The example above uses a bucket located in the region `fr-par`. If you bucket is located in another region edit the value to the corresponding region.
+    </Message>
+5. Run Hugo to generate your site for the first time
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:94]: We have 3 step 5s

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    git push -u origin main
+    ```
+
+## Setting-up the GitHub runner
+
+<Message type="tip">
+    This step requires that you have created a [GitHub repoistory](https://docs.github.com/en/get-started/quickstart/create-a-repo) for your project.
+</Message>
+
+1. Open a web browser and go to your projects' repoistory on [GitHub.com](http://github.com).
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:117]:
```suggestion
1. Open a web browser and go to your project's repository on [GitHub.com](http://github.com).
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+4. Commit the changes you made
+    ```
+    git commit -c "added a new blogpost"
+    ```
+5. Push your content to the remote repository:
+    ```
+    git push origin main
+    ```
+
+GitHub Actions launches automatically the script created in the previous script and a runner will execute Hugo, build the website and deploy it to Object Storage. You can see your updated website a few moments later in your webbrowser.
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:176]:
```suggestion
GitHub Actions automatically launches the script created in the previous step and a runner will execute Hugo, build the website and deploy it to Object Storage. You can see your updated website a few moments later in your web browser.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+For more infomration about Hugo refer the [official Hugo documentation](https://gohugo.io/about/). Furher information about GitHub actions is avialable in the [GitHub documentation](https://docs.github.com/en/actions).
```
Comment (RoRoJ) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:177]:
```suggestion
For more information about Hugo refer to the [official Hugo documentation](https://gohugo.io/about/). Further information about GitHub actions is available in the [GitHub documentation](https://docs.github.com/en/actions).
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+tags: hugo cms website github object storage
+categories: 
+  - storage
+  - object-storage
+dates:
+  validation: 2022-12-07
+  posted: 2022-12-07
+---
+
+Hugo is a static HTML and CSS website generator written in Go. The application is designed to take a directory with content and templates and render them into a full HTML website.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:17]:
```suggestion
[Hugo](https://gohugo.io/) is a static HTML and CSS website generator written in Go. The application is designed to take a directory with content and templates and render them into a full HTML website.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+It uses Markdown files with front matter for metadata, making it lightweight and easy to use.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:18]:
```suggestion
Hugo uses Markdown files with front matter for metadata, making it lightweight and easy to use. 
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+  - object-storage
+dates:
+  validation: 2022-12-07
+  posted: 2022-12-07
+---
+
+Hugo is a static HTML and CSS website generator written in Go. The application is designed to take a directory with content and templates and render them into a full HTML website.
+It uses Markdown files with front matter for metadata, making it lightweight and easy to use. 
+
+In this tutorial you learn how to install Hugo on your local computer so you can build your website locally, before pushing your content into a GitHub repository where a GitHub runner will take the content, generate the HTML version of it and upload to a Object Storage bucket automatically.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:20]:
```suggestion
In this tutorial you will learn how to install Hugo on your local computer so you can build your website locally, before pushing your content into a GitHub repository where a GitHub runner will take the content, generate the HTML version of it and upload to a Object Storage bucket automatically.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+    ```
+10. Push the content of your local repository to GitHub
+    ```
+    git push -u origin main
+    ```
+
+## Setting-up the GitHub runner
+
+<Message type="tip">
+    This step requires that you have created a [GitHub repoistory](https://docs.github.com/en/get-started/quickstart/create-a-repo) for your project.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:114]:
```suggestion
    This step requires you to have created a [GitHub repository](https://docs.github.com/en/get-started/quickstart/create-a-repo) for your project.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+</Message>
+
+1. Open a web browser and go to your project's repository on [GitHub.com](http://github.com).
+2. Click the **Settings** tab to display the settings of your repository. 
+3. Click **Secrets** > **Actions** in the side menu. The list of Actions secrets displays. GitHub Secrets are encrypted data of sensitive information. They are used here to store the Scaleway secret key and access key.
+4. Click **New repository secret** to create a new secret. Name it `SCW_ACCESS_KEY_ID` and enter your access key as secret value. Then click **Add secret** to save.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:120]:
```suggestion
4. Click **New repository secret** to create a new secret. Name it `SCW_ACCESS_KEY_ID` and enter your access key as secret value. 
5. Click **Add secret** to save your configuration.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+5. Repeat the step above with a secret named `SCW_SECRET_ACCESS_KEY` and your secret key as secret value.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:121]:
```suggestion
6. Repeat the step above with a secret named `SCW_SECRET_ACCESS_KEY` and your secret key as secret value.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+6. Click the **Actions** tab. A list of suggested GitHub Actions for your project displays. Click **Skip this and setup a workflow yourself**.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:122]:
```suggestion
7. Click the **Actions** tab. A list of suggested GitHub Actions for your project displays. Click **Skip this and setup a workflow yourself**.
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+7. Copy the follwing code in the text editor, keep the default file name `main.yml` and click **Start commit**:
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:123]:
```suggestion
8. Copy the following code in the text editor, keep the default file name `main.yml` and click **Start commit**:
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+            hugo-version: '0.108.0'
+        - name: build site
+          run: hugo
+        - name: deploy to s3
+          run: hugo deploy --force --maxDeletes -1
+          env:
+            AWS_ACCESS_KEY_ID: ${{ secrets.SCW_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY: ${{ secrets.SCW_SECRET_ACCESS_KEY }}
+    ```
+8. Enter a commit message, select **Commit directly to the `main` branch** and click **Commit new file**.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:149]:
```suggestion
9. Enter a commit message, select **Commit directly to the `main` branch** and click **Commit new file**. 
```

File: tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+          run: hugo
+        - name: deploy to s3
+          run: hugo deploy --force --maxDeletes -1
+          env:
+            AWS_ACCESS_KEY_ID: ${{ secrets.SCW_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY: ${{ secrets.SCW_SECRET_ACCESS_KEY }}
+    ```
+8. Enter a commit message, select **Commit directly to the `main` branch** and click **Commit new file**. 
+
+The runner runs automatically for the first time and, builds and deploys your website to the Object Storage bucket.
```
Comment (nedaehm) [inline tutorials/deploy-static-website-with-hugo-and-github-runners-to-object-storage/index.mdx:151]:
```suggestion
The runner runs automatically for the first time and builds and deploys your website to the Object Storage bucket.
```

**Review by remyleone — APPROVED** [review]

**Review by RoRoJ — APPROVED** [review]

**Pull request merged**